robottelo/cli/factory.py

```diff
@@ -1691,9 +1691,15 @@ def setup_org_for_a_custom_repo(options=None):
     cvv = ContentView.info({'id': cv_id})['versions'][-1]
     # Promote version to next env
     try:
-        ContentView.version_promote(
-            {'id': cvv['id'], 'organization-id': org_id, 'to-lifecycle-environment-id': env_id}
-        )
+        all_lifecycle_env = ContentView.info({'id': cv_id})['lifecycle-environments']
+        for lce in all_lifecycle_env:
+            if int(lce['id']) == env_id:
+                # already promoted to environment
+                break
+        else:
+            ContentView.version_promote(
+                {'id': cvv['id'], 'organization-id': org_id, 'to-lifecycle-environment-id': env_id}
+            )
```

Comment:
Non blocking - Not tried locally but looks more clean.
```suggestion
if not any(int(lce['id']) == env_id for lce in all_lifecycle_env):
    ContentView.version_promote(
        {'id': cvv['id'], 'organization-id': org_id, 'to-lifecycle-environment-id': env_id}
    )
```

Reply:
this looks like a mess, this might not work :-) try something like
```suggestion
       if env_id not in [lce['id'] for lce in all_lifecycle_env]:
            ContentView.version_promote(
                {'id': cvv['id'], 'organization-id': org_id, 'to-lifecycle-environment-id': env_id}
            )
```

Reply:
With above suggested approach
**Case1:**
`env_id = 7` and` lce = {['id:'3', 'name': 'A' ], [ 'id':'4', 'name': 'B' ], [ 'id':'5', 'name': 'C' ], [ 'id':'6', 'name': 'D' ]}`

CVV will promote to all lce id (i.e for lce A, B, C & D)

Reply:
sorry, I interpreted it in wrong way, it will promote to one environment for `len(lce)` times